tket/tests/test_LexiRoute.cpp

```diff
@@ -1739,4 +1740,228 @@ SCENARIO(
   mapping_frontier.circuit_.get_commands();
   REQUIRE(true);
 }
+
+SCENARIO(
+    "Test case fails if the linear boundary in merge_ancilla replaces the "
+    "ancilla entry before erasing the merge entry.") {
+  SquareGrid architecture(5, 5);
+  std::vector<Node> nodes = architecture.get_all_nodes_vec();
+  Circuit circuit(10);
+  circuit.add_op<unsigned>(OpType::CX, {0, 1});
+  circuit.add_op<unsigned>(OpType::CX, {0, 3});
+  circuit.add_op<unsigned>(OpType::CX, {1, 2});
+  circuit.add_barrier({0, 1, 2, 3, 4, 5, 6, 7, 8, 9});
+  circuit.add_op<unsigned>(OpType::CX, {6, 7});
+  std::map<Qubit, Node> p_map = {
+      // mapping for qbs with 2qb gates
+      {Qubit(0), nodes[0]},  {Qubit(1), nodes[4]},  {Qubit(2), nodes[20]},
+      {Qubit(3), nodes[24]}, {Qubit(4), nodes[11]}, {Qubit(5), nodes[17]},
+  };
+  Placement::place_with_map(circuit, p_map);
+  CompilationUnit cu(circuit);
+  PassPtr r_p = gen_routing_pass(
+      architecture, {std::make_shared<LexiLabellingMethod>(),
+                     std::make_shared<LexiRouteRoutingMethod>()});
+  REQUIRE(r_p->apply(cu));
+}
+
+SCENARIO(
+    "Test relabelling a Circuit UnitID that is an Architecture Node but "
+    "reassignable to an Ancilla Node.") {
+  GIVEN("Line Architecture, one reassignment.") {
+    std::vector<std::pair<unsigned, unsigned>> coupling_map = {
+        {0, 1}, {1, 2}, {2, 3}, {3, 4}, {4, 5}, {5, 6}, {6, 7}, {7, 8}};
+    Architecture architecture(coupling_map);
+    std::vector<Node> nodes = architecture.get_all_nodes_vec();
+    Circuit circuit(9);
+    circuit.add_op<unsigned>(OpType::CX, {0, 1});
+    for (unsigned i = 0; i < 9; i++) {
+      circuit.add_op<unsigned>(OpType::H, {i});
+    }
+    circuit.add_barrier({0, 1, 2, 3, 4, 5, 6, 7});
+    for (unsigned i = 0; i < 8; i++) {
+      circuit.add_op<unsigned>(OpType::H, {i});
+    }
+    circuit.add_op<unsigned>(OpType::CX, {0, 3});
+    circuit.add_op<unsigned>(OpType::CX, {0, 2});
+    circuit.add_op<unsigned>(OpType::CX, {0, 4});
+
+    std::map<Qubit, Node> p_map = {// mapping for qbs with 2qb gates
+                                   {Qubit(0), nodes[0]},
+                                   {Qubit(1), nodes[8]}};
+    Placement::place_with_map(circuit, p_map);
+    CompilationUnit cu(circuit);
+    PassPtr r_p = gen_routing_pass(
+        architecture, {std::make_shared<LexiLabellingMethod>(),
+                       std::make_shared<LexiRouteRoutingMethod>()});
+    REQUIRE(r_p->apply(cu));
```

Comment:
I think `REQUIRE(r_p->apply(cu));` (also in the other new tests) doesn't say much. Are there anything else we can check to prevent future regression?